doc/design/zarr-support.md

```diff
@@ -0,0 +1,100 @@
+The Zarr format is being used by NGFF to support the next generation of microscopy data. This involves a potentially
+nested folder structure with named files inside the folder. In the DANDI this would mean:
+
+1. An asset is associated with a single zarr folder. From a user perspective this is still a single asset and the UI
+   should not try to delve into the structure of the folder. The CLI should be able to download the entire tree. Matt
+   at kitware is looking into IPFS + NGFF, so we should at least keep that in mind.
+3. Blob store allows for a folder, which contains the zarr named "locations" and data. That is given a root prefix,
+   a zarr-compliant software can navigate the zarr metadata/structure using relative path rules.
```

Comment:
later we might end up with non-zarr folders. Should we include indication of underlying "folder format" within folder or the subfolder name? 
e.g. could be `d65b541b-885a-4bb4-badd-2a57b1bebab0.zarr` or `d65b541b-885a-4bb4-badd-2a57b1bebab0/zarr/`

this way we can actually support storing multiple representations for the same data withing the blob store (e.g. `KEY` - original nwb or whatnot, `KEY/zarr/` - zarr, `KEY/ipfs/` - ipfs blocks if that is a thing ;)) without causing conflicts/ambiguity when looking at a specific PREFIX and immediate "sub-folders"

Reply:
I think it would be better to store things at `zarr/KEY/` and `ipfs/KEY/`, much like we already store things in `blobs/KEY/`. We can still have the same `KEY` in multiple stores.

Reply:
Works for me. Then wording above should avoid "Blob store" since it is blobs/ for me, and mention zarr/ store

Reply:
On db/api side then it wouldn't be blobs table/endpoint right?